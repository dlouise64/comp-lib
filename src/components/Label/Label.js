import React from 'react';
import PropTypes from 'prop-types';

import './Label.css';

/** Label with required field display, htmlFor, and block styling */
function Label({htmlFor, label, required}) {
  return (
    <label style={{display: 'block'}} htmlFor={htmlFor}>
      {label} { required && <span style={{color: 'red'}}> *</span> }
    </label>
  )
}

Label.propTypes = {
  /** HTML ID for associated input */
  htmlFor: PropTypes.string.isRequired,

  /** Label text */
  label: PropTypes.string.isRequired,

  /** Display asterix after label if true */
  required: PropTypes.bool
};

Label.defaultProps = {
}

export default Label;
