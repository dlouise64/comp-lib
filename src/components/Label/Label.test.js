import React from 'react';
import renderer from 'react-test-renderer';
import Label from './Label';

test('<Label /> renders correctly', () => {
  const tree = renderer.create(<Label
    htmlFor="my_label"
    label="Sign in"
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
