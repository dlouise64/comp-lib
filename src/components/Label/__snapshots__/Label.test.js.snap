// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> renders correctly 1`] = `
<label
  className="label"
  htmlFor="my_label"
  style={
    Object {
      "display": "inline-block",
      "marginRight": "10px",
    }
  }
>
  Sign in
   
</label>
`;
